src/wagtail_headless_preview/models.py

```diff
@@ -48,17 +56,35 @@ def get_client_root_url_from_site(site):
     return root_url
 
 
-class HeadlessPreviewMixin:
+class HeadlessBase:
+    def _get_site_from_request(self, request: "HttpRequest") -> "Site":
```

Comment:
Question: why not `Site.find_for_request`?

Reply:
that was my first thought,

but then got down the rabbit hole of https://github.com/wagtail/wagtail/blob/main/wagtail/models/__init__.py#L1994-L2001 which makes use of cached root paths and checks things before falling back to the default site rather than returning None (https://github.com/wagtail/wagtail/blob/main/wagtail/models/sites.py#L162-L171)

Reply:
having said all of that, we do handle `None` in `get_client_root_url_from_site` so this shouldn't matter much, really

Reply:
changed

Reply:
That is indeed a rabbit hole :)